tests/DNSTest.php

```diff
@@ -5,42 +5,92 @@
 
 class DNSTest extends TestCase
 {
-	protected function setUp()
+	public function testConstructor()
 	{
-		set_error_handler(array($this, 'errorHandler'));
+		$d = new DNSQuery("127.0.0.1");
+		$this->assertInstanceOf('PurplePixie\\PhpDns\\DNSQuery', $d);
 	}
 
-	public function errorHandler($errno, $errstr, $errfile, $errline)
+	public function testConstructorNoServer()
+	{
+        $this->expectException(\InvalidArgumentException::class);
+		$d = new DNSQuery();
+    	}
+
+	public function testDNSQueryAndDNSAnswer()
+	{
+        $dns_server = "8.8.8.8"; // Our DNS Server
+
+        $dns_query = new DNSQuery($dns_server); // create DNS Query object - there are other options we could pass here
+		$this->assertInstanceOf('PurplePixie\\PhpDns\\DNSQuery', $dns_query);
+
+        $question = "msn.com"; // the question we will ask
+        $type = "A"; // the type of response(s) we want for this question
+
+        $result = $dns_query->query($question, $type); // do the query
+		$this->assertInstanceOf('PurplePixie\\PhpDns\\DNSAnswer', $result);
+
+        //Process Results
+        $count = $result->count(); // number of results returned
+        $this->assertEquals(1, $count);
+
+        foreach ($result as $result_count) {
+            // only after A records
+            if ($result_count->getType() === "A") {
+                $this->assertEquals(1, $result_count->getTypeId());
+                $this->assertEquals('13.82.28.61', $result_count->getData());
+                $this->assertEquals('msn.com has IPv4 address 13.82.28.61', $result_count->getString());
+                $this->assertEquals(1, count($result_count->getExtras()));
```

Comment:
It has the `assertCount` assert function to assert the array count is expected.

This `assertCount` function usage is as follows:

```php
......
 $this->assertCount(1, $result_count->getExtras());
......
```